### PR TITLE
feat: Add support bundle spec to velero addon 1.12.2

### PR DESCRIPTION
**Pull request by banjoh**
#### What this PR does / why we need it:
Install a velero support bundle spec whenever velero addon is added to a kURL spec.

#### Which issue(s) this PR fixes:

Fixes [sc-67291](https://app.shortcut.com/replicated/stories/space/67291/cre)

#### Does this PR introduce a user-facing change?

```release-note
Install a velero support bundle spec whenever velero addon is added to a kURL spec
```

### CODE REVIEW

**Review by laverya — APPROVED** [review]

File: addons/velero/1.12.2/tmpl-troubleshoot.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: kurl-velero-supportbundle-spec
+  labels:
+    troubleshoot.io/kind: support-bundle
```
Comment (laverya) [inline addons/velero/1.12.2/tmpl-troubleshoot.yaml:6]: you're creating this in the default namespace

either include this in the files applied by kustomize, or include the namespace here
I would recommend including this in the files applied by kustomize

**Review by laverya — CHANGES_REQUESTED** [review]

File: addons/velero/1.12.2/tmpl-troubleshoot.yaml (hunk 1 above)
Comment (banjoh) [reply addons/velero/1.12.2/tmpl-troubleshoot.yaml:6]: Good point. I'll create the config maps in the same namespace as the addons.

Comment (banjoh) [reply addons/velero/1.12.2/tmpl-troubleshoot.yaml:6]: @laverya I made the requested change

Comment (laverya) [reply addons/velero/1.12.2/tmpl-troubleshoot.yaml:6]: Almost good to approve - you need to add the kustomization change to the template files

File: addons/velero/1.9.5/tmpl-kustomization.yaml [hunk 2]
```diff
@@ -2,3 +2,4 @@ namespace: ${VELERO_NAMESPACE}
 
 resources:
 - velero.yaml
+- troubleshoot.yaml
```
[the remark this answers is not in the document]
Comment (banjoh) [reply addons/velero/1.9.5/tmpl-kustomization.yaml]: @laverya change made

**Review by laverya — APPROVED** [review]

**Pull request merged**